src/platform_impl/windows/window_state.rs

```diff
@@ -344,6 +349,21 @@ impl WindowFlags {
             }
         }
 
+        if diff.contains(WindowFlags::CLOSABLE) || new.contains(WindowFlags::CLOSABLE) {
+            unsafe {
+                EnableMenuItem(
+                    GetSystemMenu(window, 0),
+                    SC_CLOSE,
+                    MF_BYCOMMAND
+                        | if new.contains(WindowFlags::CLOSABLE) {
+                            MF_ENABLED
+                        } else {
+                            MF_DISABLED
+                        },
+                );
```

Comment:
Could we compose this on a separate line, so this is a bit more readable?

Reply:
separating it on a new line didn't feel to me as much enhancement so I too the liberty to use `bool::then` in [`4ff0733`](https://github.com/rust-windowing/winit/pull/2537/commits/4ff07333d7a68fdf655996e5962dcc7c8c1c579f)